providers/linux/host_linux.go

```diff
@@ -81,6 +81,30 @@ func (h *host) VMStat() (*types.VMStatInfo, error) {
 	return parseVMStat(content)
 }
 
+// NetworkCounters reports data from /proc/net on linux
+func (h *host) NetworkCounters() (*types.NetworkCountersInfo, error) {
+
+	snmpRaw, err := ioutil.ReadFile(h.procFS.path("net/snmp"))
+	if err != nil {
+		return nil, err
+	}
+	snmp, err := getNetSnmpStats(snmpRaw)
+	if err != nil {
+		return &types.NetworkCountersInfo{}, err
```

Comment:
I think this should return a nil value on error. It seem inconsistent to have it both ways.